File: backend/model/request/vps.go
package request

type VPSEditable struct {
	MerchantID        string `json:"merchant_id" binding:"required"`
	Code              string `json:"code" binding:"required"`
	Name              string `json:"name" binding:"required"`
	Description       string `json:"description"`
	CPUCores          int    `json:"cpu_cores"`
	MemoryMB          int    `json:"memory_mb"`
	DiskGB            int    `json:"disk_gb"`
	DiskType          string `json:"disk_type"`
	TransferGB        *int   `json:"transfer_gb"`
	PortMbps          *int   `json:"port_mbps"`
	HasIPv4           bool   `json:"has_ipv4"`
	IPv4Count         int    `json:"ipv4_count"`
	HasIPv6           bool   `json:"has_ipv6"`
	IPv6Count         int    `json:"ipv6_count"`
	PriceAmount       string `json:"price_amount"`
	Currency          string `json:"currency"`
	BillingPeriod     string `json:"billing_period"`
	PurchaseURL       string `json:"purchase_url"`
	Enabled           bool   `json:"enabled"`
	CollectionEnabled bool   `json:"collection_enabled"`
}
type VPSCreate struct{ VPSEditable }
type VPSUpdate struct {
	ID string `json:"id" binding:"required"`
	VPSEditable
}
type VPSListQuery struct {
	Page          int    `form:"page"`
	PageSize      int    `form:"page_size"`
	Q             string `form:"q"`
	MerchantID    string `form:"merchant_id"`
	Currency      string `form:"currency"`
	BillingPeriod string `form:"billing_period"`
	Status        *int   `form:"status"`
	Sort          string `form:"sort"`
	Enabled       *bool  `form:"enabled"`
}

File: frontend/src/types/vps.ts
// VPS 相关类型契约
import { ID, Timestamp, PageQuery } from './common';
import { Merchant } from './merchant';
import { Stock, StockStatus } from './stock';

export type BillingPeriod = 'monthly' | 'quarterly' | 'yearly' | 'one_time';
export type DiskType = 'ssd' | 'nvme' | 'hdd' | 'unknown';
export type VPSSortOption = 'updated_desc' | 'price_asc' | 'price_desc';

export interface VPS {
  id: ID;
  merchant: Merchant;
  code: string;
  name: string;
  description: string;
  cpu_cores: number;
  memory_mb: number;
  disk_gb: number;
  disk_type: DiskType;
  transfer_gb: number | null;
  port_mbps: number | null;
  has_ipv4: boolean;
  ipv4_count: number;
  has_ipv6: boolean;
  ipv6_count: number;
  price_amount: string;
  currency: string;
  billing_period: BillingPeriod;
  purchase_url: string;
  stock: Stock;
  created_at: Timestamp;
  updated_at: Timestamp;
}

export interface AdminVPS extends VPS {
  merchant_id: ID;
  enabled: boolean;
  collection_enabled: boolean;
}

export type Vps = VPS;
export type VpsAdmin = AdminVPS;

export interface VPSQuery extends PageQuery {
  q?: string;
  merchant_id?: string;
  currency?: string;
  billing_period?: BillingPeriod;
  status?: StockStatus;
  sort?: VPSSortOption;
}

export interface AdminVPSQuery extends VPSQuery {
  enabled?: boolean;
}

export interface VPSCreateInput {
  merchant_id: ID;
  code: string;
  name: string;
  description: string;
  cpu_cores: number;
  memory_mb: number;
  disk_gb: number;
  disk_type: DiskType | string;
  transfer_gb?: number | null;
  port_mbps?: number | null;
  has_ipv4: boolean;
  ipv4_count: number;
  has_ipv6: boolean;
  ipv6_count: number;
  price_amount: string;
  currency: string;
  billing_period: BillingPeriod;
  purchase_url: string;
  enabled: boolean;
  collection_enabled: boolean;
}

export interface VPSUpdateInput extends VPSCreateInput {
  id: ID;
}
